cf/data/data.py

```diff
@@ -8028,143 +7329,76 @@ def mean(
-        >>> print(e.array)
-        [[[ 0  1  2  3]
-          [ 4  5  6  7]]
-         [[ 8  9 10 11]
-          [12 13 14 15]]
-         [[16 17 18 19]
-          [20 21 22 23]]]
-
-        >>> e.mean(axes=[0, 2])
-        <CF Data(1, 2, 1): [[[9.5, 13.5]]]>
-        >>> f = e.mean(axes=[0, 2], squeeze=True)
-        >>> f
-        <CF Data(2): [9.5, 13.5]>
-        >>> f.shape
-        (2,)
-        >>> print(e.mean(axes=[0, 1]).array)
-        [[[10. 11. 12. 13.]]]
-        >>> print(e.mean(axes=[0, 1], weights={(1, 0): w}).array)
-        [[[11. 12. 13. 14.]]]
-
-        >>> e[0, 0] = cf.masked
-        >>> e[-1, -1] = cf.masked
-        >>> e[..., 2] = cf.masked
-        >>> print(e.array)
-        [[[-- -- -- --]
-          [4 5 -- 7]]
-         [[8 9 -- 11]
-          [12 13 -- 15]]
-         [[16 17 -- 19]
-          [-- -- -- --]]]
-
-        >>> e.mean()
-        <CF Data(1, 1, 1): [[[11.333333333333334]]]>
-        >>> print(e.mean(axes=[0, 1]).array)
-        [[[10.0 11.0 -- 13.0]]]
-        >>> print(e.mean(axes=[0, 1], weights={(1, 0): w}).array)
-        [[[9.666666666666666 10.666666666666666 -- 12.666666666666666]]]
-
-        """
-        return self._collapse(
-            mean_f,
-            mean_fpartial,
-            mean_ffinalise,
-            axes=axes,
-            squeeze=squeeze,
+        <CF Data(1, 1): [[5.878787878787879]] K>
+
+        """
+        d = _inplace_enabled_define_and_cleanup(self)
+        d, _ = _collapse(
+            Collapse.mean,
+            d,
+            axis=axes,
             weights=weights,
+            keepdims=not squeeze,
+            split_every=split_every,
             mtol=mtol,
-            inplace=inplace,
-            _preserve_partitions=_preserve_partitions,
         )
+        return d
 
+    @daskified(_DASKIFIED_VERBOSE)
+    @_inplace_enabled(default=False)
     def mean_absolute_value(
         self,
         axes=None,
         squeeze=False,
         mtol=1,
         weights=None,
+        split_every=None,
         inplace=False,
-        _preserve_partitions=False,
     ):
-        """Collapse axes with their mean absolute value.
+        """Calculate mean absolute values.
 
-        Missing data elements are omitted from the calculation.
+        Calculates the mean absolute value or the mean absolute values
+        along axes.
 
-        .. seealso:: `maximum`, `minimum`, `mean`, `mid_range`, `sum`, `sd`,
-                     `var`
+        See
+        https://ncas-cms.github.io/cf-python/analysis.html#collapse-methods
+        for mathematical definitions.
+
+         ..seealso:: `sample_size`, `mean`, `sd`, `sum`
 
         :Parameters:
 
-            axes : (sequence of) int, optional
+            {{collapse axes: (sequence of) `int`, optional}}
+
+            {{weights: data_like, `dict`, or `None`, optional}}
+
+            {{collapse squeeze: `bool`, optional}}
+
+            {{mtol: number, optional}
```

Comment:
```suggestion
            {{mtol: number, optional}}
```

There's about twenty cases of this throughout `data.data` here, as I see from a `grep`. I won't pose them all as review 'suggestions'! I suggest addressing them in batch via `sed` or similar...